src/agent/transform-headers-agent.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-underscore-dangle */
+const http = require('http');
+const WrappedAgent = require('./wrapped-agent');
+
+const { _storeHeader } = http.OutgoingMessage.prototype;
+
+class TransformHeadersAgent extends WrappedAgent {
+    transformRequest(request) {
```

Comment:
We may not have been completely thorough with this ourselves, but until we move to TypeScript, please use JSDoc comments with all functions like this example:

```
    /**
     * @param {string} key - proxy host unique key
     * @returns {string} - http version
     */
```